airflow/jobs.py

```diff
@@ -990,11 +989,14 @@ def _execute_task_instances(self,
         # Get all the queued task instances from associated with scheduled
         # DagRuns.
         TI = models.TaskInstance
+        DR = models.DagRun
         task_instances_to_examine = (
             session
             .query(TI)
             .filter(TI.dag_id.in_(simple_dag_bag.dag_ids))
             .filter(TI.state.in_(states))
+            .outerjoin(DR, and_(DR.dag_id==TI.dag_id, DR.execution_date==TI.execution_date))
+            .filter(or_(DR.run_id == None, not_(DR.run_id.like('%backfill%'))))
```

Comment:
It's a different predicate. Otherwise the statement returns NULL which gets filtered